src/legacy/core_plugins/kbn_doc_views/public/views/table/table_row.tsx

```diff
@@ -0,0 +1,110 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { ReactNode } from 'react';
+import { FieldName } from 'ui/directives/field_name/field_name';
+import { FieldMapping, DocViewFilterFn } from 'ui/registry/doc_views_types';
+import classNames from 'classnames';
+import { DocViewTableRowBtnFilterAdd } from './table_row_btn_filter_add';
+import { DocViewTableRowBtnFilterRemove } from './table_row_btn_filter_remove';
+import { DocViewTableRowBtnToggleColumn } from './table_row_btn_toggle_column';
+import { DocViewTableRowBtnCollapse } from './table_row_btn_collapse';
+import { DocViewTableRowBtnFilterExists } from './table_row_btn_filter_exists';
+import { DocViewTableRowIconNoMapping } from './table_row_icon_no_mapping';
+import { DocViewTableRowIconUnderscore } from './table_row_icon_underscore';
+
+export interface Props {
+  field: string;
+  fieldMapping?: FieldMapping;
+  displayNoMappingWarning: boolean;
+  displayUnderscoreWarning: boolean;
+  isCollapsible: boolean;
+  isColumnActive: boolean;
+  isCollapsed: boolean;
+  onToggleCollapse: () => void;
+  onFilter?: DocViewFilterFn;
+  onToggleColumn?: () => void;
+  value: string | ReactNode;
+  valueRaw: unknown;
+}
+
+export function DocViewTableRow({
+  field,
+  fieldMapping,
+  displayNoMappingWarning,
+  displayUnderscoreWarning,
+  isCollapsible,
+  isCollapsed,
+  isColumnActive,
+  onFilter,
+  onToggleCollapse,
+  onToggleColumn,
+  value,
+  valueRaw,
+}: Props) {
+  const valueClassName = classNames({
+    kbnDocViewer__value: true,
+    'truncate-by-height': isCollapsible && isCollapsed,
+  });
+
+  return (
+    <tr key={field} data-test-subj={`tableDocViewRow-${field}`}>
+      {typeof onFilter === 'function' && (
+        <td className="kbnDocViewer__buttons">
+          <DocViewTableRowBtnFilterAdd
+            disabled={!fieldMapping || !fieldMapping.filterable}
+            onClick={() => onFilter(fieldMapping, valueRaw, '+')}
+          />
+          <DocViewTableRowBtnFilterRemove
+            disabled={!fieldMapping || !fieldMapping.filterable}
+            onClick={() => onFilter(fieldMapping, valueRaw, '-')}
+          />
+          {typeof onToggleColumn === 'function' && (
+            <DocViewTableRowBtnToggleColumn active={isColumnActive} onClick={onToggleColumn} />
+          )}
+          <DocViewTableRowBtnFilterExists
+            disabled={!fieldMapping || !fieldMapping.filterable}
+            onClick={() => onFilter('_exists_', field, '+')}
+            scripted={fieldMapping && fieldMapping.scripted}
+          />
+        </td>
+      )}
+      <td className="kbnDocViewer__field">
+        <FieldName field={fieldMapping} fieldName={field}></FieldName>
+      </td>
+      {isCollapsible ? (
```

Comment:
nit: Seems like the only difference between the branches is the `DocViewTableRowBtnCollapse` element, would it make sense to pull the condition into the `td`? `{isCollapsible && <DocViewTableRowBtnCollapse onClick={onToggleCollapse} isCollapsed={isCollapsed} />}`

Reply:
yes, thx!
done!